core/usage_interceptor.go

```diff
@@ -38,9 +39,13 @@ func unaryServerInterceptor(pre preFunc, post postFunc) grpc.UnaryServerIntercep
 		}
 		res, err := handler(newCtx, req)
 		if err != nil {
+			perr := post(newCtx, info.FullMethod)
+			if perr != nil {
+				log.Errorf("unaryServerInterceptor postFunc: %v", err)
+			}
```

Comment:
In the event there's an error returned from the gRPC handler, we still want to call the post usage function that submits usage info from the context.